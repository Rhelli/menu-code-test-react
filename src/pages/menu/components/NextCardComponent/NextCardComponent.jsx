import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import styles from './NextCardComponent.module.scss';

const NextCardComponent = ({ checkOrderFinalisation }) => {
  const history = useHistory();
  const handleBackButton = () => history.push('/');

  return (
    <div className={styles.nextCardContainer}>
      <button onClick={handleBackButton} className={styles.previousButton}>
        Previous
      </button>
      <button onClick={checkOrderFinalisation} className={styles.nextButton}>
        Next
      </button>
    </div>
  );
};

NextCardComponent.propTypes = {
  checkOrderFinalisation: PropTypes.func.isRequired
};

export default NextCardComponent;
