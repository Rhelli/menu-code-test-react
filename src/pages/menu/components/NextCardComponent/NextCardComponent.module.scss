.nextCardContainer {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-evenly;
  align-items: center;
  height: 10%;
  margin-top: 20px;
  width: 100%;
  padding: 20px;
  border: 2px solid #707070;
  border-radius: 4px;

  .previousButton,
  .nextButton {
    height: 40px;
    width: 40%;
    outline: none;
    border: none;
    border-radius: 4px;
    color: #fff;
  }

  .previousButton {
    background-color: #da3743;
    border-bottom: 2px solid #a42932;
    transition-duration: 0.1s;

    &:hover {
      background-color: #bf303b;
      border-bottom: 1px solid #88222a;
      transition-duration: 0.1s;
    }
  }

  .nextButton {
    background-color: #8bc34a;
    border-bottom: 2px solid #689238;
    transition-duration: 0.1s;

    &:hover {
      background-color: #7aab41;
      border-bottom: 1px solid #577a2e;
      transition-duration: 0.1s;
    }
  }
}
