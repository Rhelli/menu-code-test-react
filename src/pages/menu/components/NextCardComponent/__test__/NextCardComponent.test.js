import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import NextCardComponent from '../NextCardComponent';

Enzyme.configure({ adapter: new Adapter() });

const checkOrderFinalisation = jest.fn();
const nextCardComponent = (<NextCardComponent checkOrderFinalisation={checkOrderFinalisation} />);

describe('The appearance and functionality of the NextCardComponent', () => {
  const container = mount(<NextCardComponent checkOrderFinalisation={checkOrderFinalisation} />);
  it('Matches the previous snapshot', () => {
    const tree = renderer.create(nextCardComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly', () => {
    mount(<NextCardComponent checkOrderFinalisation={checkOrderFinalisation} />);
  });

  it('Has the correct structure', () => {
    expect(container.childAt(0).hasClass('nextCardContainer')).toBe(true);
    expect(container.childAt(0).children()).toHaveLength(2);
    expect(container.childAt(0).childAt(0).name()).toBe('button');
    expect(container.childAt(0).childAt(1).name()).toBe('button');
  });

  it('Triggers the order check when the next button is clicked', () => {
    container.find('.nextButton').simulate('click');
    expect(checkOrderFinalisation).toHaveBeenCalledTimes(1);
  });
});
