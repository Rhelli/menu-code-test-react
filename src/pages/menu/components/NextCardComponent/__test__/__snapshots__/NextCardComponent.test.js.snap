// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the NextCardComponent Matches the previous snapshot 1`] = `
<div
  className="nextCardContainer"
>
  <button
    className="previousButton"
    onClick={[Function]}
  >
    Previous
  </button>
  <button
    className="nextButton"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
